tests/integration/targets/aci_aep_to_epg/tasks/main.yml

```diff
@@ -0,0 +1,215 @@
+    state: present
+
+- name: Configure VRF
+  cisco.aci.aci_vrf:
+    <<: *tenant_present
+    vrf: epg2aep
+
+- name: Configure BD
+  cisco.aci.aci_bd:
+    <<: *tenant_present
+    vrf: epg2aep
+    bd: epg2aep
+
+- name: Configure Ap
+  cisco.aci.aci_ap:
+    <<: *tenant_present
+    ap: ap
+
+- name: Configure EPG
+  cisco.aci.aci_epg:
+    <<: *tenant_present
+    ap: ap
+    epg: '{{ item }}'
+    bd: epg2aep
+  loop:
+    - epg1
+    - epg2
+
+# ADD ASSOCIATION
+- name: Add EPG to AEP association (check_mode)
+  cisco.aci.aci_aep_to_epg: &assoc_present
+    <<: *aep_present
+    tenant: epg2aep
+    ap: ap
+    epg: epg1
+    encap: 1984
+    interface_mode: trunk
+  check_mode: yes
+  register: cm_add_assoc
+
+- name: Add EPG to AEP association (normal mode)
+  cisco.aci.aci_aep_to_epg:
+    <<: *assoc_present
+  register: nm_add_assoc
+
+- name: Verify add_assoc
+  assert:
+    that:
+    - cm_add_assoc is changed
+    - nm_add_assoc is changed
+    - cm_add_assoc.previous == nm_add_assoc.previous == []
+    - cm_add_assoc.sent.infraGeneric.children[0].infraRsFuncToEpg.attributes.encap == nm_add_assoc.sent.infraGeneric.children[0].infraRsFuncToEpg.attributes.encap == 'vlan-1984'
+    - cm_add_assoc.sent.infraGeneric.children[0].infraRsFuncToEpg.attributes.mode == nm_add_assoc.sent.infraGeneric.children[0].infraRsFuncToEpg.attributes.mode == 'regular'
+    - cm_add_assoc.sent.infraGeneric.children[0].infraRsFuncToEpg.attributes.tDn == nm_add_assoc.sent.infraGeneric.children[0].infraRsFuncToEpg.attributes.tDn == 'uni/tn-epg2aep/ap-ap/epg-epg1'
+
+- name: Add EPG to AEP association again, check if idempotency works (check_mode)
+  cisco.aci.aci_aep_to_epg:
+    <<: *assoc_present
+  check_mode: yes
+  register: cm_add_assoc_again
+
+- name: Add EPG to AEP association again, check if idempotency works (normal_mode)
+  cisco.aci.aci_aep_to_epg:
+    <<: *assoc_present
+  register: nm_add_assoc_again
+
+- name: Verify add_assoc_again
+  assert:
+    that:
+    - cm_add_assoc_again is not changed
+    - nm_add_assoc_again is not changed
+
+# UPDATE ASSOCIATION
+- name: Change encap (check_mode)
+  cisco.aci.aci_aep_to_epg: &assoc_update
+    <<: *assoc_present
+    encap: 1985
+    primary_encap: 1986
+  check_mode: yes
+  register: cm_update_assoc
+
+- name: Change encap (normal_mode)
+  cisco.aci.aci_aep_to_epg:
+    <<: *assoc_update
+  register: nm_update_assoc
+
+- name: Verify update_assoc
+  assert:
+    that:
+    - cm_update_assoc is changed
+    - nm_update_assoc is changed
+    - cm_update_assoc.previous == nm_update_assoc.previous != []
+    - cm_update_assoc.sent.infraGeneric.children[0].infraRsFuncToEpg.attributes.encap == nm_update_assoc.sent.infraGeneric.children[0].infraRsFuncToEpg.attributes.encap == 'vlan-1985'
+    - cm_update_assoc.sent.infraGeneric.children[0].infraRsFuncToEpg.attributes.primary_encap == nm_update_assoc.sent.infraGeneric.children[0].infraRsFuncToEpg.attributes.primary_encap == 'vlan-1986'
```

Comment:
primary_encap -> primaryEncap

Reply:
Sure, stupid mistake, sorry about that ;-)